projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php

```diff
@@ -222,7 +222,20 @@ public function add_tools_menu( $wp_admin_import = false, $wp_admin_export = fal
 	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
 	 */
 	public function add_options_menu( $wp_admin = false ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
-		add_menu_page( esc_attr__( 'Settings', 'jetpack' ), __( 'Settings', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/general/' . $this->domain, null, 'dashicons-admin-settings', 80 );
+		$slug = 'https://wordpress.com/settings/general/' . $this->domain;
+		add_menu_page( esc_attr__( 'Settings', 'jetpack' ), __( 'Settings', 'jetpack' ), 'manage_options', $slug, null, 'dashicons-admin-settings', 80 );
+		add_submenu_page( $slug, esc_attr__( 'General', 'jetpack' ), __( 'General', 'jetpack' ), 'manage_options', $slug );
+		add_submenu_page( $slug, esc_attr__( 'Security', 'jetpack' ), __( 'Security', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/security/' . $this->domain );
+		add_submenu_page( $slug, esc_attr__( 'Performance', 'jetpack' ), __( 'Performance', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/performance/' . $this->domain );
```

Comment:
Small nitpick, can we extract this into a separate method, like we did with `add_posts_menu`? We can pass `$slug` as an argument and also reuse it in class-admin-menu.php:433

Reply:
I think this is one of these cases where I prefer to sacrifice maintainability over readability and consistency. I don't like maintaining duplicated fragments of code, but it's something I'm willing to accept if the resulting code of avoiding it ends up being inconsistent with how the rest of the code behaves and/or forces the developer to jump back-and-forth into multiple source files in order have a fully understanding of the logic.

However, if folks feel strongly against this, I'm happy to refactor the code and get rid of the duplications.

Reply:
I prefer to have them in just one place because at some point the arguments can change for various reasons and we might end up with a menu with some arguments on one env while having different arguments on another one, after that things can be a bit tricky to change by someone else because they might not know the context of that change. 

I don't have strong feelings about it since the changes are minor. :)